Templates/Kotlin/Operation.stencil

```diff
@@ -13,7 +13,7 @@ import javax.annotation.Generated
 
 @Generated("{% if moduleName %}{{ moduleName }}{% else %}com.shopify.syrup{% endif %}")
 {% endif %}
-class {{ name }}({{ operation.variables|renderKotlinArguments }}): {% if isQuery %}Query{% else %}Mutation{% endif %}<{{ operation.name }}Response> {
+class {{ name }}({{ operation.variables|renderKotlinArguments }}): {% if isQuery %}Query{% else %}{{ operation.typeName }}{% endif %}<{{ operation.name }}Response> {
 
     override val rawQueryString = "{{ queryString|replace:"$","\$"|replaceQuotes }}"
 
```

Comment:
I don't think `subscriptions` will ever need `selections` since it won't use caching. so below on line 28 we might want to check the `operation.typeName` to decide wether to render

Reply:
and similarly i think `operationVariables` is only used for selections. so both can be defaulted to empty map and list respectively for subs

Reply:
What are `selections` in this context? Could you give an example?

Reply:
selections are a nested map of fields used to deconstruct the response for caching

Reply:
```
    override val rawQueryString = "query Shop { __typename shop { __typename name } }"

    override val selections = listOf<Selection>(
        Selection(
            name = "shop",
            type = "Shop",
            cacheKey = "shop",
            passedGID = null,
            typeCondition = "QueryRoot",
            shouldSkipBasedOnConditionalDirective = false,
            selections = listOf<Selection>(
                Selection(
                    name = "name",
                    type = "String",
                    cacheKey = "name",
                    passedGID = null,
                    typeCondition = "Shop",
                    shouldSkipBasedOnConditionalDirective = false,
                    selections = listOf<Selection>()
                )
            )
        )
    )
}
```

Reply:
If we're concerned about volume of code generated, lets enhance the configuration flag `generateSelections` to specify which operations to generate them for, not just hardcode it out for subscriptions specifically. Other client implementations may want to have selections available on their subscription operations.